bundles/CoreBundle/Migrations/Version20201007000000.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\CoreBundle\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+use Symfony\Component\Cache\Adapter\DoctrineDbalAdapter;
+
+/**
+ * @internal
+ */
+final class Version20201007000000 extends AbstractMigration
+{
+    /**
+     * @param Schema $schema
+     */
+    public function up(Schema $schema): void
+    {
+        if (!$schema->hasTable('cache_items')) {
+            $cacheAdapter = new DoctrineDbalAdapter(\Pimcore\Db::get());
```

Comment:
@BlackbitDevs I think we should use dependency injection here:
```suggestion
            $cacheAdapter = $this->container->get('pimcore.cache.adapter.doctrine_dbal');
```
For this you must implement the `Symfony\Component\DependencyInjection\ContainerAwareInterface` and use the `Symfony\Component\DependencyInjection\ContainerAwareTrait`

Reply:
Actually I wanted to keep it similar to https://github.com/pimcore/pimcore/blob/955d733b826ea588adabe64d52c883ae65ea274b/bundles/InstallBundle/src/Installer.php#L565-L566
Otherwise it may be difficult to see if there is a corresponding migration for a command of `Installer`. Or is using dependency injection not possible in the `Installer`?

Reply:
It is fine as is...`pimcore.cache.adapter.doctrine_dbal` is not public.